terraform/modules/cluster/mig-with-container/cloudinit/templates/userdata-gpu.yaml.template

```diff
@@ -33,23 +33,44 @@ ${aiinfra_pull_image}
   content: |
     #!/bin/sh
     docker_device_flags=$(find /dev -type c -regex '\/dev\/nvidia[0-9]*' -printf '--device %p:%p ')
-    docker run \
-      --detach \
-      --hostname $(hostname) \
-      --ipc host \
-      --name aiinfra \
-      --network host \
-      --privileged \
-      --restart always \
-      --volume /var/lib/nvidia/lib64:/usr/local/nvidia/lib64 \
-      --volume /var/lib/nvidia/bin:/usr/local/nvidia/bin \
-      --device /dev/nvidia-uvm:/dev/nvidia-uvm \
-      --device /dev/nvidiactl:/dev/nvidiactl \
-      $${docker_device_flags} \
-      ${docker_volume_flags} \
-      ${docker_env_flags} \
-      ${docker_options} \
-      ${docker_image} ${docker_cmd}
+    if [ ${enable_cloud_logging} ]; then
```

Comment:
This is going to become very unsustainable very quickly as more `enable_`s are added. im thinking we should combine this with docker_options and have the `container` variable look something like:
```terraform
variable "container" {
  cmd = string // maybe list(string)?
  image = string
  run_options = object({
    env = map(string)
    enable_cloud_logging = bool
    extra = list(string)
  })
}
```
and leave most trivial options to `extra` (so we don't have to explicitly write each and every `docker run` option)